kairos-deposit-contract/compile-contracts.sh

```diff
@@ -0,0 +1,16 @@
+rustup target add wasm32-unknown-unknown
+
+cargo build -p contract --release --target wasm32-unknown-unknown
+wasm-opt --strip-debug --signext-lowering ./target/wasm32-unknown-unknown/release/deposit-contract.wasm -o ./target/wasm32-unknown-unknown/release/deposit-contract-optimized.wasm
```

Comment:
This introduces `wasm-opt` dependency that has to be manually installed :unamused:.

I believe we can remove it from this bash script, and include [`wasm_opt` crate](https://docs.rs/wasm-opt/0.116.0/wasm_opt/) in the tests, that will be used for creating optimized file side-by-side (unfortunately the library does not allow working directly on memory).

Reply:
Installing wasm-opt system wide is a viable option imo. Anyone working with casper contracts should have it available anyways. Makefiles in other contracts rely on it as well and it is as long as it works I don't see a need to change anything about it.

Reply:
>  Anyone working with casper contracts should have it available anyways.

@jonas089 Wrong. **You** have it installed, because I showed you this can be used for [fixing opcode 192 issue](https://github.com/casper-network/casper-node/issues/4367#issuecomment-1798438389) and using most recent Rust complier. By default Casper contracts rely on `wasm-strip` (binary from `wabt`), so I am still insisting on using crate instead of unknown system binary.

Reply:
Since you know exactly what to do here and I don't, would it be possible for you to submit a PR over this issue? Otherwise I can look into it later today.